src/core/datatypes/phylogenetics/geography/RegionalFeatures.cpp

```diff
@@ -0,0 +1,280 @@
+//
+//  RegionalFeatures.cpp
+//  rb_mlandis
+//
+//  Created by Michael Landis on 12/3/13.
+//  Copyright (c) 2013 Michael Landis. All rights reserved.
+//
+
+#include "RegionalFeatures.h"
+#include "RegionalFeatureLayer.h"
+#include "RbConstants.h"
+
+#include <cmath>
+#include <iostream>
+#include <map>
+#include <vector>
+#include <sstream>
+#include <string>
+
+using namespace RevBayesCore;
+
+
+RegionalFeatures::RegionalFeatures(void) {
+    ;
+}
+
+
+RegionalFeatures::RegionalFeatures(std::map<size_t, std::map<size_t, std::vector<long> > > wc,
+                                   std::map<size_t, std::map<size_t, std::vector<double> > > wq,
+                                   std::map<size_t, std::map<size_t, std::vector<std::vector<long> > > > bc,
+                                   std::map<size_t, std::map<size_t, std::vector<std::vector<double> > > > bq) :
+    withinCategorical(wc),
+    withinQuantitative(wq),
+    betweenCategorical(bc),
+    betweenQuantitative(bq)
+{
+    
+    initializeFeatures();
+}
+
+RegionalFeatures::RegionalFeatures(const RegionalFeatures& a)
+{
+    if (this != &a) {
+        numLayers = a.numLayers;
+        numTimeslices = a.numTimeslices;
+        withinCategorical = a.withinCategorical;
+        withinQuantitative = a.withinQuantitative;
+        betweenCategorical = a.betweenCategorical;
+        betweenQuantitative = a.betweenQuantitative;
+        feature_layers = a.feature_layers;
+        *this = a;
```

Comment:
Looks like you are BOTH delegating to operator= and ALSO copying all the fields.

Maybe just delegate to `operator=( )`?